myapp/spec/models/user_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validaton' do
+    it 'create user successfully' do
+      user = build(:user)
+      expect(user).to be_valid
+    end
+
+    it 'no error messages when user created successfully' do
+      user = build(:user)
+      user.valid?
+      expect(user.errors).to be_empty
+    end
+
+    it 'user cannot be created without username' do
+      user = build(:user, username: '')
+      expect(user).to be_invalid
+    end
+
+    it 'show error messages if username is empty' do
+      user = build(:user, username: '')
+      user.valid?
+      expect(user.errors[:username]).to eq ["can't be blank"]
+    end
```

Comment:
how about using `context`?
such as
```
context 'when username is empty' do
  before do
    user = build(:user, username: '')
  end

  it 'cannot be created' do
  end

  it 'shows error message' do
  end
end
```

Reply:
And this helps us to reduce repetition `build(:user, username: '')`.

Reply:
Changed the format . Thank you.